compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/StreamingCodeDesugar.java

```diff
@@ -0,0 +1,812 @@
+        eventObjectField.fieldKind = FieldKind.SINGLE;
+        eventObjectField.pos = selectClause.pos;
+        eventObjectField.field = ASTBuilderUtil.createIdentifier(selectClause.pos, "eventObject");
+
+        BLangExpression typeCastingExpression = generateConversionExpr(eventObjectField, inputStreamEventType,
+                symResolver);
+
+        BLangVariable typeCastingVariable = ASTBuilderUtil.
+                createVariable(selectClause.pos, getVariableName(lambdaFunctionVariable.getName().getValue()),
+                        inputStreamEventType, typeCastingExpression, typeCastingVarSymbol);
+
+        typeCastingVariable.typeNode = ASTBuilderUtil.createTypeNode(inputStreamEventType);
+        BLangVariableDef typeCastingVariableDef = ASTBuilderUtil.createVariableDef(selectClause.pos,
+                typeCastingVariable);
+        lambdaBody.stmts.add(typeCastingVariableDef);
+
+        //Output object creation
+        BLangRecordLiteral outputEventRecordLiteral = ASTBuilderUtil.createEmptyRecordLiteral(selectClause.pos,
+                outputEventTypeVariable.type);
+        List<BLangRecordLiteral.BLangRecordKeyValue> recordKeyValueList = new ArrayList<>();
+
+        List<? extends SelectExpressionNode> selectExprList = selectClause.getSelectExpressions();
+        for (SelectExpressionNode expressionNode : selectExprList) {
+            BLangSelectExpression selectExpression = (BLangSelectExpression) expressionNode;
+            BLangRecordLiteral.BLangRecordKeyValue recordKeyValue = (BLangRecordLiteral.BLangRecordKeyValue)
+                    TreeBuilder.createRecordKeyValue();
+
+            if (selectExpression.getIdentifier() != null) {
+                BLangSimpleVarRef varRef = (BLangSimpleVarRef) TreeBuilder.createSimpleVariableReferenceNode();
+                varRef.variableName = ASTBuilderUtil.createIdentifier(selectClause.pos,
+                        selectExpression.getIdentifier());
+                recordKeyValue.key = new BLangRecordLiteral.BLangRecordKey(varRef);
+                recordKeyValue.key.fieldSymbol = getOutputEventFieldSymbol(outputEventTypeVariable,
+                        selectExpression.getIdentifier());
+            } else {
+                BLangSimpleVarRef varRef = (BLangSimpleVarRef) TreeBuilder.createSimpleVariableReferenceNode();
+                varRef.variableName = ((BLangFieldBasedAccess) selectExpression.getExpression()).field;
+                recordKeyValue.key = new BLangRecordLiteral.BLangRecordKey(varRef);
+                recordKeyValue.key.fieldSymbol = getOutputEventFieldSymbol(outputEventTypeVariable,
+                        ((BLangFieldBasedAccess) selectExpression.getExpression()).field.value);
+            }
+
+            if (selectExpression.getExpression() instanceof BLangFieldBasedAccess) {
+                BLangFieldBasedAccess fieldAccessExpr = ((BLangFieldBasedAccess) selectExpression.getExpression());
```

Comment:
Is a clone of the field access expression created here? If so, you may use the existing method `cloneAccessExpr(BLangAccessExpression originalAccessExpr)` in Desugar class.

Reply:
It is not a close of the field based expression. Rather we are creating another expression using the newly created var symbol.